__std__/cells/plutus/packages/check-the-flake.nix

```diff
@@ -1,8 +1,7 @@
 { inputs, cell }:
 
 # TODO(std) path must be fixed once __std__ is brought to the toplevel.
-# TODO(std) make this part of CI
-# TODO(std) turn this into a single derivation using recurseForDerivations
```

Comment:
I really find this script super useful the way it is.
It has helped me find bugs all over the place.
It also implicitly verifies the `one-derivation-per-file` rule and the `file-name == flake-fragment-name` rule.
I think turning it into a normal derivation using `recurseForDerivations` & co would make it worse.

Reply:
`nix flake check` has this problem https://github.com/NixOS/nix/issues/4265
Running `nix flake check --system x86_64-darwin --show-trace` (both with and without the `--system` flag) returns :
```
error: a 'x86_64-linux' with features {} is required to build '/nix/store/b8z3p4b25dv1dc756852b63kk6n63l2y-cabal.project.drv', but I am a 'x86-64_darwin' with features {benchmark, big-parallel, nixos-test}

       … while realising the context of path '/nix/store/m76radakffd5n56x1iq15zj8jbkd063f-haskell-project-plan-to-nix-pkgs'

       at /nix/store/z0g3r63vhf8h1ac2fmcv7bygz5llnxmz-source/lib/import-and-filter-project.nix:18:13:

           17|   projectSubDir'' = if projectSubDir == "" then "" else projectSubDir + "/"; # With trailing /
           18|   project = import "${projectNix}${projectSubDir'}";
             |             ^
           19| in project // {


[...]


       … while checking the derivation 'devShells.x86_64-linux.default'

       at /nix/store/ldhr3ysw5vmry7iggzrv832bai80v080-cells/plutus/devshells/default.nix:9:3:

            8|
            9|   default = plutus-shell;
             |   ^
           10| }

       … while checking flake output 'devShells'
``` 

Odd that it's trying to check `x86_64-linux` even on Darwin and even with the `--system` flag 🤦 
The `--no-build` flag also does nothing ...